storage/scylla_api.c

```diff
@@ -450,6 +514,109 @@ status_t init_scylla(CassCluster** cluster, CassSession* session, char* hosts, b
   free(use_query);
   return ret;
 }
+
+static CassStatement* ret_insert_identity_statement(const CassPrepared* prepared, const identity_table_t* obj) {
+  CassStatement* statement = NULL;
+  statement = cass_prepared_bind(prepared);
+  cass_statement_bind_uuid_by_name(statement, "code", ret_identity_uuid(obj));
+  cass_statement_bind_bytes_by_name(statement, "hash", ret_identity_hash(obj), NUM_FLEX_TRITS_HASH);
+  cass_statement_bind_int8_by_name(statement, "status", ret_identity_status(obj));
+  return statement;
+}
+
+status_t insert_identity_table(CassSession* session, identity_table_t* obj) {
+  status_t ret = SC_OK;
+  const CassPrepared* insert_prepared = NULL;
+  CassStatement* statement = NULL;
+  const char* insert_query =
+      "INSERT INTO identity (code, status, hash)"
+      "VALUES (?, ?, ?);";
+  if (session == NULL || obj == NULL) {
+    ta_log_error("%s\n", "SC_TA_NULL");
+    return SC_TA_NULL;
+  }
+
+  if (prepare_query(session, insert_query, &insert_prepared) != CASS_OK) {
+    ta_log_error("%s\n", "prepare INSERT query fail");
+    return SC_STORAGE_CASSANDRA_QUREY_FAIL;
+  }
+  statement = ret_insert_identity_statement(insert_prepared, obj);
+  if (execute_statement(session, statement) != CASS_OK) {
+    ta_log_error("%s\n", "SC_STORAGE_CASSANDRA_QUREY_FAIL");
+    ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
+    goto exit;
+  }
+
+exit:
+  cass_prepared_free(insert_prepared);
+  return ret;
+}
+
+static status_t get_identity_array(CassSession* session, CassStatement* statement, identity_array_t* identity_array) {
+  status_t ret = SC_OK;
+  CassFuture* future = NULL;
+  const CassResult* result;
+  CassIterator* iterator;
+  identity_table_t* identity;
+  new_identity(&identity);
+  future = cass_session_execute(session, statement);
+  cass_future_wait(future);
+  if (cass_future_error_code(future) != CASS_OK) {
+    print_error(future);
+    ret = SC_STORAGE_CASSANDRA_QUREY_FAIL;
+    goto exit;
+  }
+
+  result = cass_future_get_result(future);
+  iterator = cass_iterator_from_result(result);
+
+  while (cass_iterator_next(iterator)) {
+    identity_table_t* itr;
+    const cass_byte_t* hash;
+    cass_int8_t value;
+    CassUuid uuid;
+    size_t len;
```

Comment:
Would it be more efficient to declare the variables above out of the loop?

Reply:
Declaring inside the loop helps to minimize the scope (only visible in this block scope).